tools/src/main/scala/com/nvidia/spark/rapids/tool/planparser/SQLPlanParser.scala

```diff
@@ -36,7 +36,9 @@ class ExecInfo(
     val isSupported: Boolean,
     val children: Option[Seq[ExecInfo]], // only one level deep
     val stages: Set[Int] = Set.empty,
-    val shouldRemove: Boolean = false) {
+    val shouldRemove: Boolean = false,
+    val unsupportedExecs: String = "",
```

Comment:
this needs to be a string for the WholeStageCodegen?  Could we simply turn this to boolean and then get exec names from the children?

Reply:
Sorry, I didn't get your question. I am keeping unsupportedExecs as String irrespective of whether the exec is in WholeStageCodeGen or outside of it.

Reply:
The reason I am keeping it String is to store the "exec" name to QualificationAppInfo. In WholeStageCodeGen, there could be multiple Execs and we are creating string which contains all the execnames and passing it to ExecInfo class. If we change it to boolean, then getting all the exec names later will not be possible. Please let me know if I am missing anything here.

Reply:
>The reason I am keeping it String is to store the "exec" name to QualificationAppInfo. 

I'm not sure what you mean by this. The aggregateStats is the thing that parseSQlPlan and create the PlanInfo.  Inside of that we have the ExecInfo and each ExecInfo already has the isSupported flag and the exec name.

Actually you don't need this field at all, right?  Since each ExecInfo already has an isSupported Boolean field.  so all you have to do is search for that and then get "exec" val for the name.  Now for wholeStageCodeGen you just look at the "children" field. You actually already have code that is really close:

```
val unSupportedExecsArray = childNodes.filter(_.unsupportedExecs.nonEmpty).map(
      x => x.exec)
```


This just needs to be:
```
val unSupportedExecsArray = childNodes.filter(!_.isSupported).map(
      x => x.exec)

```

The original line is actually weird because you don't even use the unsupportedExecs field, you just get the name from "exec" field.  Instead of doing that inside of the WholeStageExecParser we just do that when we create the QualificationSummaryInfo.

For instance (you can probably clean this up) but something like:

 ```
 val unSupportedExecs2 = origPlanInfos.flatMap { p =>
        val topLevelExecs = p.execInfo.filter(!_.isSupported)
        val childrenExecs = p.execInfo.flatMap { e =>
          e.children.map(x => x.filter(!_.isSupported))
        }.flatten
        topLevelExecs ++ childrenExecs
      }.map(_.exec).toSet.mkString(";").trim
```

We don't have any children with children so you don't have to worry about recursing

Reply:
Thanks for the explanation. It makes sense. I have updated the code to reflect this change.